### PR TITLE
Add new Windows/Schannel-only flags.

### PR DESCRIPTION
**Pull request by anrossi**
## Description

Add new Schannel-only flags to support certain scenarios. Completes #2573.

## Testing

Updated existing certificate flag test.

## Documentation

Updated the flags documentation.

### CODE REVIEW

File: src/inc/msquic.h [hunk 1]
```diff
@@ -109,7 +109,10 @@ typedef enum QUIC_CREDENTIAL_FLAGS {
     QUIC_CREDENTIAL_FLAG_IGNORE_REVOCATION_OFFLINE              = 0x00001000, // Schannel only currently
     QUIC_CREDENTIAL_FLAG_SET_ALLOWED_CIPHER_SUITES              = 0x00002000,
     QUIC_CREDENTIAL_FLAG_USE_PORTABLE_CERTIFICATES              = 0x00004000,
-    QUIC_CREDENTIAL_FLAG_USE_SUPPLIED_CREDENTIALS               = 0x00008000, // Schannel only currently
+    QUIC_CREDENTIAL_FLAG_USE_SUPPLIED_CREDENTIALS               = 0x00008000, // Schannel only
+    QUIC_CREDENTIAL_FLAG_USE_SYSTEM_MAPPER                      = 0x00010000, // Schannel only
+    QUIC_CREDENTIAL_FLAG_CHAIN_CACHE_ONLY_URL_RETRIEVAL         = 0x00020000, // Schannel only currently
```
Comment (Techwolfy) [inline src/inc/msquic.h:114]: `_CHAIN_` should be excluded here to match the other flags.

File: src/platform/tls_schannel.c [hunk 2]
```diff
@@ -1052,11 +1053,20 @@ CxPlatTlsSecConfigCreate(
     if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_IGNORE_REVOCATION_OFFLINE) {
         Credentials->dwFlags |= SCH_CRED_IGNORE_REVOCATION_OFFLINE;
     }
+    if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_CHAIN_CACHE_ONLY_URL_RETRIEVAL) {
+        Credentials->dwFlags |= SCH_CRED_CACHE_ONLY_URL_RETRIEVAL;
+    }
+    if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_REVOCATION_CHECK_CACHE_ONLY) {
+        Credentials->dwFlags |= SCH_CRED_REVOCATION_CHECK_CACHE_ONLY;
+        CERT_CHAIN_CACHE_ONLY_URL_RETRIEVAL ;
```
Comment (Techwolfy) [inline src/platform/tls_schannel.c:1061]: Copy/paste error?

**Review by Techwolfy — CHANGES_REQUESTED** [review]

File: src/platform/tls_schannel.c (hunk 2 above)
Comment (anrossi) [reply src/platform/tls_schannel.c:1061]: whoops! I was using that to jump-to-definition.

File: src/inc/msquic.h (hunk 1 above)
Comment (anrossi) [reply src/inc/msquic.h]: Agreed.

**Review by Techwolfy — APPROVED** [review]

File: src/inc/msquic.h [hunk 3]
```diff
@@ -109,7 +109,10 @@ typedef enum QUIC_CREDENTIAL_FLAGS {
     QUIC_CREDENTIAL_FLAG_IGNORE_REVOCATION_OFFLINE              = 0x00001000, // Schannel only currently
     QUIC_CREDENTIAL_FLAG_SET_ALLOWED_CIPHER_SUITES              = 0x00002000,
     QUIC_CREDENTIAL_FLAG_USE_PORTABLE_CERTIFICATES              = 0x00004000,
-    QUIC_CREDENTIAL_FLAG_USE_SUPPLIED_CREDENTIALS               = 0x00008000, // Schannel only currently
+    QUIC_CREDENTIAL_FLAG_USE_SUPPLIED_CREDENTIALS               = 0x00008000, // Schannel only
+    QUIC_CREDENTIAL_FLAG_USE_SYSTEM_MAPPER                      = 0x00010000, // Schannel only
+    QUIC_CREDENTIAL_FLAG_CACHE_ONLY_URL_RETRIEVAL               = 0x00020000, // Schannel only currently
+    QUIC_CREDENTIAL_FLAG_REVOCATION_CHECK_CACHE_ONLY            = 0x00040000, // Schannel only currently
```
Comment (nibanks) [inline src/inc/msquic.h:115]: Your documentation above says "Windows only". Is it "Windows only" or "Schannel only"? I am guessing Windows.

**Review by thhous-msft — APPROVED** [review]

**Review by nibanks — APPROVED** [review]

**Pull request merged**